packages/react/src/components/participant/BarVisualizer.tsx

```diff
@@ -23,7 +24,9 @@ export interface BarVisualizerProps extends React.HTMLProps<HTMLDivElement> {
   state?: AgentState;
   /** Number of bars that show up in the visualizer */
   barCount?: number;
+  /** @deprecated use `track` field instead */
   trackRef?: TrackReferenceOrPlaceholder;
+  track?: TrackReferenceOrPlaceholder | LocalAudioTrack | RemoteAudioTrack;
```

Comment:
suggestion: Other components currently seem to favor the `trackRef` prop name rather than `track`. I think I understand why you did it here (as a stepping stone to migrating away from `TrackReference`) but this means this component has two similar props which could be used at the same time, resulting in a less clear end interface.

What about instead, making `BarVisualizerProps` actually have two versions in a discriminated union, one that is the pre-existing version with `trackRef` (plus adding the deprecation), and the other getting rid of `trackRef` and adding `track` instead? The nice thing about doing it that way is the fields are mutually exclusive.

Reply:
To be clear I don't really have a strong opinion on this, it just struck me reading this code that it would be nice if they were mutually exclusive since they really should never both be set and then once you've migrated over to the new `track` prop, it is as if the old `trackRef` prop no longer exists .

By "props wouldn't be as easy to inspect anymore" if you mean that the component props wouldn't point to a singular interface on a "go to definition" type operation and instead would be an expression of some sort - yes, that is a good point. I guess there are pros and cons to both approaches, so do whatever you think makes sense.